pkg/tmpl/context_funcs.go

```diff
@@ -286,11 +287,17 @@ func ToYaml(v interface{}) (string, error) {
 }
 
 func FromYaml(str string) (Values, error) {
-	m := Values{}
+	m := map[string]interface{}{}
 
 	if err := yaml.Unmarshal([]byte(str), &m); err != nil {
 		return nil, fmt.Errorf("%s, offending yaml: %s", err, str)
 	}
+
+	m, err := maputil.CastKeysToStrings(m)
```

Comment:
https://github.com/helmfile/helmfile/pull/609/files#r1058925709

This is especially important here because this function is supposed to be used like `fromYaml | toJson` within helmfile templates. Go's `json` pkg which powers `toJson` is unable to serialize `map[interface{}]interface{}` deserialized by yaml.v2. This CastKeysToStrings call normalizes it so that both yaml.v2 and goccy/go-yaml implementation of FromYaml is compatible with ToJson.